packages/lesswrong/components/users/Auth0PasswordResetPage.tsx

```diff
@@ -0,0 +1,148 @@
+    width: 52,
+  },
+  title: {
+    fontSize: 24,
+    fontWeight: 600,
+  },
+  subtitle: {
+    color: theme.palette.grey[800]
+  },
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '12px',
+    width: '100%',
+  },
+  inputContainer: {
+    display: 'flex',
+    alignItems: 'center',
+    gap: '12px',
+    borderRadius: theme.borderRadius.default,
+    background: theme.palette.panelBackground.loginInput,
+    padding: '0 17px',
+  },
+  input: {
+    flexGrow: 1,
+    padding: '15px 0px',
+    color: theme.palette.grey[1000],
+    background: 'transparent',
+    fontSize: 14,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    fontWeight: 500,
+    '&::placeholder': {
+      color: theme.palette.grey[600],
+    },
+  },
+  message: {
+    fontSize: 14,
+    fontWeight: 500,
+    color:  theme.palette.grey[1000],
+  },
+  error: {
+    fontSize: 14,
+    fontWeight: 500,
+    color: theme.palette.text.error2,
+  },
+  button: {
+    width: '100%',
+    height: 50,
+    padding: '15px 17px',
+    fontWeight: 600,
+  },
+});
+
+const Auth0PasswordResetPage = ({ classes }: { classes: ClassesType<typeof styles> }) => {
+  const { EAButton, SingleColumnSection } = Components;
+  const { query } = useLocation();
+  const client = useAuth0Client();
+
+  const [email, setEmail] = useState('');
+  const [message, setMessage] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const emailFromQuery = query.email;
+    if (emailFromQuery) {
+      setEmail(emailFromQuery);
+    }
+  }, [query]);
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setMessage(null);
+    setError(null);
+    try {
+      await client.resetPassword(email);
+      setMessage('Password reset email sent');
+    } catch (e) {
+      setError('Error sending password reset email');
+    }
+  };
+
+  return (
+    <SingleColumnSection className={classes.root}>
+      <div className={classes.content}>
+        <div className={classes.header}>
+          <div className={classes.lightbulb}>{lightbulbIcon}</div>
+          <h1 className={classes.title}>Set password</h1>
+          <p className={classes.subtitle}>Submit the form below to receive a password reset email.</p>
+        </div>
+        <form onSubmit={handleSubmit} className={classes.form}>
+          <div className={classes.inputContainer}>
+            <input
+              type="email"
+              className={classes.input}
+              placeholder="Email"
+              value={email}
+              onChange={(e) => setEmail(e.target.value)}
+              required
+            />
+          </div>
+          {message && <div className={classes.message}>{message}</div>}
+          {error && <div className={classes.error}>{error}</div>}
+          <EAButton type="submit" style="primary" className={classes.button}>
+            Submit
+          </EAButton>
```

Comment:
The reason I'm making this a form rather than a link that triggers a password reset directly is to avoid accidentally sending multiple password resets on refresh. Also I think this will be more robust as we want it to be the case that people can set a password long after Facebook login has been removed